uniffi_build/src/lib.rs

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use anyhow::{bail, Result};
+use std::{env, process::Command};
+
+pub fn generate_scaffolding(idl_file: &str) -> Result<()> {
+    println!("cargo:rerun-if-changed={}", idl_file);
+    // Why don't we just depend on uniffi-bindgen and call the public functions?
+    // Calling the command line helps making sure that the generated swift/Kotlin/whatever
+    // bindings were generated with the same version of uniffi as the Rust scaffolding code.
+    let out_dir = env::var("OUT_DIR").map_err(|_| anyhow::anyhow!("$OUT_DIR missing?!"))?;
+    let status = Command::new("uniffi-bindgen")
```

Comment:
I'm a little conflicted here, I like `uniffi-bindgen` being a separate CLI crate, but the way its setup requires us to run the scaffolding through a command when in reality this mostly will be run on build scripts (Unless we think the consumer should to be able to generate their scaffolding at different times, then I concede)

If we think that it's okay for the consumer to **always** use `uniffi_build` to generate scaffolding (and never `uniffi_bindgen`. Then maybe all the scaffolding logic can be in this crate?

What do you think? I'm genuinely unsure, so okay with whatever path you choose!

If we choose to stay with the way that it is, it would be nice if we can print a nice error message or something if the user doesn't have `uniffi-bindgen` installed

Reply:
> (Unless we think the consumer should to be able to generate their scaffolding at different times, then I concede)

I think this is at least partly the intention, for consumers who want to checkin the generated scaffolding rather than generating it afresh each time (like we currently do for protobuf generated rust code in appservices).

Reply:
Yes that's the intent, my concern is that just like `prost`, compiling this crate at build time will make the build take way longer. This way you have the flexibility of doing that, or like @rfk said you can generate the scaffolding out-of-build.